internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -810,45 +816,93 @@ func ExpandSiteConfigWindowsWebAppSlot(siteConfig []SiteConfigWindowsWebAppSlot,
 	}
 
 	if metadata.ResourceData.HasChange("site_config.0.always_on") {
-		expanded.AlwaysOn = utils.Bool(winSlotSiteConfig.AlwaysOn)
+		expanded.AlwaysOn = pointer.To(winSlotSiteConfig.AlwaysOn)
 	}
 
 	if metadata.ResourceData.HasChange("site_config.0.api_management_api_id") {
 		expanded.APIManagementConfig = &web.APIManagementConfig{
-			ID: utils.String(winSlotSiteConfig.ApiManagementConfigId),
+			ID: pointer.To(winSlotSiteConfig.ApiManagementConfigId),
 		}
 	}
 
 	if metadata.ResourceData.HasChange("site_config.0.api_definition_url") {
 		expanded.APIDefinition = &web.APIDefinitionInfo{
-			URL: utils.String(winSlotSiteConfig.ApiDefinition),
+			URL: pointer.To(winSlotSiteConfig.ApiDefinition),
 		}
 	}
 
 	if metadata.ResourceData.HasChange("site_config.0.app_command_line") {
-		expanded.AppCommandLine = utils.String(winSlotSiteConfig.AppCommandLine)
+		expanded.AppCommandLine = pointer.To(winSlotSiteConfig.AppCommandLine)
 	}
 
 	if metadata.ResourceData.HasChange("site_config.0.application_stack") {
 		if len(winSlotSiteConfig.ApplicationStack) == 1 {
 			winAppStack := winSlotSiteConfig.ApplicationStack[0]
-			expanded.NetFrameworkVersion = utils.String(winAppStack.NetFrameworkVersion)
-			expanded.PhpVersion = utils.String(winAppStack.PhpVersion)
-			expanded.NodeVersion = utils.String(winAppStack.NodeVersion)
-			expanded.PythonVersion = utils.String(winAppStack.PythonVersion)
-			expanded.JavaVersion = utils.String(winAppStack.JavaVersion)
-			expanded.JavaContainer = utils.String(winAppStack.JavaContainer)
-			expanded.JavaContainerVersion = utils.String(winAppStack.JavaContainerVersion)
-			if winAppStack.DockerContainerName != "" {
+
+			if winAppStack.NetFrameworkVersion != "" {
+				expanded.NetFrameworkVersion = pointer.To(winAppStack.NetFrameworkVersion)
+				if currentStack == "" {
+					currentStack = CurrentStackDotNet
+				}
+			}
+			if winAppStack.NetCoreVersion != "" {
+				expanded.NetFrameworkVersion = pointer.To(winAppStack.NetFrameworkVersion)
+				if currentStack == "" {
+					currentStack = CurrentStackDotNetCore
+				}
+			}
+			if winAppStack.NodeVersion != "" {
+				// Note: node version is now exclusively controlled via app_setting.WEBSITE_NODE_DEFAULT_VERSION
+				if currentStack == "" {
+					currentStack = CurrentStackNode
+				}
+			}
+			if winAppStack.PhpVersion != "" {
+				if winAppStack.PhpVersion != PhpVersionOff {
+					expanded.PhpVersion = pointer.To(winAppStack.PhpVersion)
+				} else {
+					expanded.PhpVersion = pointer.To("")
+				}
+				if currentStack == "" {
+					currentStack = CurrentStackPhp
+				}
+			}
+			if winAppStack.PythonVersion != "" || winAppStack.Python {
+				expanded.PythonVersion = pointer.To(winAppStack.PythonVersion)
```

Comment:
Could something weird happen here if `winAppStack.PythonVersion` is `""` and `winAppStack.Python` is true. Do we want to do this instead?

```
if winAppStack.PythonVersion != "" && winAppStack.Python {
```

Reply:
The second setting is superseding the first, however, the service no longer uses the value at all. This is here to catch the deprecated property and ensure that the `current_stack` value is set appropriately for the metadata in the resource so the portal displays the correct view for the application stack blade.